src/modules/seasons/seasons.controller.ts

```diff
@@ -1,12 +1,34 @@
-import { Controller, Get } from '@nestjs/common';
+import { 
+    Body, 
+    Controller, 
+    Get, 
+    Param, 
+    Post, 
+    UsePipes, 
+    ValidationPipe 
+} from '@nestjs/common';
 import { SeasonsService } from './seasons.service';
+import { CreateSeasonDto, FindOneSeasonParam, SeasonDto } from './seasons.types';
 
 @Controller('seasons')
 export class SeasonsController {
     constructor(private readonly seasonsService: SeasonsService) {}
 
     @Get()
-    getSeasons(): object {
+    async getSeasons(): Promise<SeasonDto[]> {
         return this.seasonsService.findAll();
     }
+
+    @Get(':id')
+    @UsePipes(new ValidationPipe())
+    async getOneSeason(@Param() params: FindOneSeasonParam): Promise<SeasonDto | object> {
+        const res = await this.seasonsService.findOne(params.id);
+        return res ? res : { message: "We couldn't find a season with that id!"}
```

Comment:
Great job thinking about errors here. there is a better way to return errors with NestJS though.

There are a few different types we can use. In this case, there should be a `NotFoundException`error class we can throw here, and NestJS will handle responding in a proper error format with the correct error code (404).